lang/java/avro/src/test/java/org/apache/avro/generic/TestGenericData.java

```diff
@@ -198,6 +198,25 @@ public void testMapValuesEquals() {
     assertEquals(r1, r0);
   }
 
+  @Test
+  public void testArrayValuesEqualsStringAndUtf8Compatibility() {
+    Schema arrayElementSchema = new Schema.Parser().parse("{\"type\": \"string\"}");
```

Comment:
Not 100% if this is what you mean, but looking at it again I do see that the code can be simplified as:

```
    Field myMapField = new Field("my_map", Schema.createMap(Schema.create(Schema.Type.STRING)), null, null);
    Schema schema = Schema.createRecord("my_record", "doc", "mytest", false);
```

and 

```
    Field myArrayField = new Field("my_array", Schema.createArray(Schema.create(Schema.Type.STRING)), null, null);
    Schema schema = Schema.createRecord("my_record", "doc", "mytest", false);
```

I can force-push that if you'd prefer... or am I missing something else?

Reply:
```
Field myMapField = new Field("my_map", Schema.createMap(Schema.create(Schema.Type.STRING)), null, null);
Schema schema = Schema.createRecord("my_record", "doc", "mytest", false);
```
Yes, I think this one is more correct (or less confusing). Otherwise, as it was before, it creates nested maps (map in map). Not a big deal since finally it still tests maps comparison.